`404 Not Found: @vjpr/geojson-utils0.1.0 (via https://registry.npmjs.org/@vjpr%2Fgeojson-utils0.1.0)`

_package.json_

```
    "geojson-utils": "npm:@vjpr/geojson-utils0.1.0",
```

_pnpm-debug.log_

```
   err: 
      name: "Error"
      message: "404 Not Found: @vjpr/geojson-utils0.1.0 (via https://registry.npmjs.org/@vjpr%2Fgeojson-utils0.1.0)"
      code: "ERR_PNPM_REGISTRY_META_RESPONSE_404"
      stack: "Error: 404 Not Found: @vjpr/geojson-utils0.1.0 (via https://registry.npmjs.org/@vjpr%2Fgeojson-utils0.1.0)\n    at /Users/Vaughan/nvm/versions/node/v8.9.0/pnpm-global/1/node_modules/.registry.npmjs.org/pnpm/2.13.1/node_modules/pnpm/lib/node_modules/@pnpm/npm-resolver/lib/pickPackage.js:103:25\n    at Generator.next (<anonymous>)\n    at fulfilled (/Users/Vaughan/nvm/versions/node/v8.9.0/pnpm-global/1/node_modules/.registry.npmjs.org/pnpm/2.13.1/node_modules/pnpm/lib/node_modules/@pnpm/npm-resolver/lib/pickPackage.js:4:58)\n    at <anonymous>\n    at process._tickCallback (internal/process/next_tick.js:188:7)"
```

If I make a fresh new repo I don't see this error.

Looks like the url is messed up. https://registry.npmjs.org/@vjpr%2Fgeojson-utils0.1.0

Probably something to do with parsing aliases with `@scopes`.

---

EDIT: I am seeing same issue without scopes.

Maybe a general issue with aliases.

### pnpm version:2.13.1

--- a/packages/npm-resolver/src/fetch.ts
+++ b/packages/npm-resolver/src/fetch.ts
@@ -13,19 +13,22 @@ class RegistryResponseError extends PnpmError {
   public readonly response: RegistryResponse
   public readonly uri: string
 
-  constructor (opts: {
+  constructor (info: string, opts: {
     package: string,
     response: RegistryResponse,
     uri: string,
   }) {
     super(
       `REGISTRY_META_RESPONSE_${opts.response.status}`,
-      `${opts.response.status} ${opts.response.statusText}: ${opts.package} (via ${opts.uri})`)
+      `${opts.response.status} ${opts.response.statusText}: ${opts.package} (via ${opts.uri})${info}`)
     this.package = opts.package
     this.response = opts.response
     this.uri = opts.uri
   }
 }
+
+// https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
+const semvarRegex = new RegExp(/(.*)(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$/)
 
 export default async function fromRegistry (
   fetch: (url: string, opts: {auth?: object}) => Promise<{}>,
@@ -36,7 +39,12 @@ export default async function fromRegistry (
   const uri = toUri(pkgName, registry)
   const response = await fetch(uri, { auth }) as RegistryResponse
   if (response.status > 400) {
-    throw new RegistryResponseError({
+    let info = ''
+    const matched = pkgName.match(semvarRegex)
+    if (matched) {
+      info = ` Did you mean ${matched[1]}?`
+    }
+    throw new RegistryResponseError(info, {
       package: pkgName,
       response,
       uri,

--- a/packages/npm-resolver/src/fetch.ts
+++ b/packages/npm-resolver/src/fetch.ts
@@ -8,16 +8,24 @@ type RegistryResponse = {
   json: () => Promise<PackageMeta>,
 }
 
+// https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
+const semvarRegex = new RegExp(/(.*)(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$/)
+
 class RegistryResponseError extends PnpmError {
   public readonly package: string
   public readonly response: RegistryResponse
   public readonly uri: string
 
-  constructor (info: string, opts: {
+  constructor (opts: {
     package: string,
     response: RegistryResponse,
     uri: string,
   }) {
+    let info = ''
+    const matched = opts.package.match(semvarRegex)
+    if (matched) {
+      info = ` Did you mean ${matched[1]}?`
+    }
     super(
       `REGISTRY_META_RESPONSE_${opts.response.status}`,
       `${opts.response.status} ${opts.response.statusText}: ${opts.package} (via ${opts.uri})${info}`)
@@ -26,9 +34,6 @@ class RegistryResponseError extends PnpmError {
     this.uri = opts.uri
   }
 }
-
-// https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
-const semvarRegex = new RegExp(/(.*)(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$/)
 
 export default async function fromRegistry (
   fetch: (url: string, opts: {auth?: object}) => Promise<{}>,
@@ -39,12 +44,7 @@ export default async function fromRegistry (
   const uri = toUri(pkgName, registry)
   const response = await fetch(uri, { auth }) as RegistryResponse
   if (response.status > 400) {
-    let info = ''
-    const matched = pkgName.match(semvarRegex)
-    if (matched) {
-      info = ` Did you mean ${matched[1]}?`
-    }
-    throw new RegistryResponseError(info, {
+    throw new RegistryResponseError({
       package: pkgName,
       response,
       uri,

--- a/packages/npm-resolver/test/index.ts
+++ b/packages/npm-resolver/test/index.ts
@@ -720,6 +720,30 @@ test('error is thrown when package is not found in the registry', async t => {
   }
 })
 
+test('extra info is shown if package has valid semver appended', async t => {
+  const notExistingPackage = 'foo1.0.0'
+
+  nock(registry)
+    .get(`/${notExistingPackage}`)
+    .reply(404, {})
+
+  const resolveFromNpm = createResolveFromNpm({
+    metaCache: new Map(),
+    rawConfig: { registry },
+    storeDir: tempy.directory(),
+  })
+  try {
+    await resolveFromNpm({ alias: notExistingPackage, pref: '1.0.0' }, { registry })
+    t.fail('installation should have failed')
+  } catch (err) {
+    t.equal(err.message, `404 Not Found: ${notExistingPackage} (via https://registry.npmjs.org/foo1.0.0) Did you mean foo?`)
+    t.equal(err['package'], notExistingPackage)
+    t.equal(err['code'], 'ERR_PNPM_REGISTRY_META_RESPONSE_404')
+    t.equal(err['uri'], `${registry}${notExistingPackage}`)
+    t.end()
+  }
+})
+
 test('error is thrown when there is no package found for the requested version', async t => {
   nock(registry)
     .get('/is-positive')

